sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/unresolved.scala

```diff
@@ -392,16 +392,27 @@ case class UnresolvedStar(target: Option[Seq[String]]) extends Star with Unevalu
           throw QueryCompilationErrors.starExpandDataTypeNotSupportedError(target.get)
       }
     } else {
-      val from = input.inputSet.map(_.name).mkString(", ")
-      val targetString = target.get.mkString(".")
-      throw QueryCompilationErrors.cannotResolveStarExpandGivenInputColumnsError(
-        targetString, from)
+      // Change the unresolved star to an expanded star to indicate the star has been
+      // expanded once.
+      ExpandedStar(this) :: Nil
     }
   }
 
   override def toString: String = target.map(_ + ".").getOrElse("") + "*"
 }
 
+/**
+ * Represent an [[UnresolvedStar]] that has been to expand once.
```

Comment:
`has been to expand once` sounds uncertain definition to me. By its definition, cannot it be expanded more than once?

Reply:
This `ExpandedStar` is indeed confusing. It was used to detect invalid usage of star expressions. I will update the logic here.